source/agora/flash/Node.d

```diff
@@ -1292,6 +1295,19 @@ public abstract class FlashNode : FlashControlAPI
 
         return true;
     }
+
+    ///
+    protected FeeUTXOs getFeeUTXOs (PublicKey pk, ulong tx_size)
+    {
+        auto per_byte = this.listener.getEstimatedTxFee();
+        if(!per_byte.mul(tx_size))
+            return FeeUTXOs.init;
+        auto utxos = this.listener.getFeeUTXOs(pk, per_byte);
+        // reuse total_value as refund amount
+        if (!utxos.total_value.sub(per_byte))
+            utxos.total_value = Amount(0);
+        return utxos;
+    }
```

Comment:
Why do we have two implementations of this ?

Reply:
this one takes `tx_size`
`FlashListener.getFeeUTXOs` takes flat `Amount`